### PR TITLE
Phase2-hgx263 Try to resolve the orientation index issue for HGCal V13/V14

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Try to resolve the orientation index issue for HGCal V13/V14

#### PR validation:

Tested using runTheMatrix workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

File: Geometry/HGCalCommonData/src/HGCalWaferMask.cc [hunk 1]
```diff
@@ -333,6 +333,42 @@ bool HGCalWaferMask::goodCell(int u, int v, int n, int type, int rotn) {
   return good;
 }
 
+int HGCalWaferMask::getRotation(int zside, int type, int rotn) {
+  int newrotn(rotn);
+  if ((zside < 0) && (type != HGCalTypes::WaferFull)) {
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalWaferMask.cc:338]: should you check if `rotn` is less than `6` for these cases?
(could also make `6` a shared, named constant)

Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalWaferMask.cc:338]: Done and committed

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/src/HGCalWaferMask.cc [hunk 2]
```diff
@@ -333,6 +333,44 @@ bool HGCalWaferMask::goodCell(int u, int v, int n, int type, int rotn) {
   return good;
 }
 
+int HGCalWaferMask::getRotation(int zside, int type, int rotn) {
+  if (rotn >= HGCalTypes::WaferCornerMax)
+    rotn = HGCalTypes::WaferCorner0;
+  int newrotn(rotn);
+  if ((zside < 0) && (type != HGCalTypes::WaferFull)) {
+    if (type == HGCalTypes::WaferFive) {  //WaferFive
+      static const int rot1[6] = {HGCalTypes::WaferCorner4,
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalWaferMask.cc:342]: @bsunanda do all the `6`s here correspond to `HGCalTypes::WaferCornerMax`? if so, please replace them

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/src/HGCalWaferMask.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalWaferMask.cc]: I believe they are already replaced. Can you approve this PR now?

**Comment by kpedro88** [comment]
@bsunanda I just noticed some problems reported by the static checks:
```
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-08-1100/src/Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalHEFileAlgo.cc:503:70: error: use of undeclared identifier 'wafers_'
          edm::LogVerbatim("HGCalGeom") << " DDHGCalHEFileAlgo: " << wafers_[type] << " number " << copy << " type "
                                                                     ^
1 error generated.
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-08-1100/src/Geometry/HGCalCommonData/src/HGCalGeomParameters.cc:1965:7: warning: Value stored to 'n' is never read
      n = 0;
      ^   ~
1 warning generated.
```

**Comment by bsunanda** [comment]
@kpedro88 Can you approve this PR now - I need it for something else. I am sending a separate PR for the static checks

**Comment by kpedro88** [comment]
+upgrade

**Comment by bsunanda** [comment]
@cvuosalo @silviodonato Can you approve this PR and merge it at the earliest?

**Comment by silviodonato** [comment]
merge

**Comment by cvuosalo** [comment]
+1

**Pull request merged**